bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/gemini/gems/Canopus.java

```diff
@@ -384,27 +247,21 @@ private Area probe3DependentRange(ObsContext ctx, Point2D.Double mid, Tuple2<Dou
         // that positive y is down and a positive rotation rotates the positive
         // x axis toward the positive y axis.  Position angle is expressed as
         // an angle east of north.
-        double t = ctx.getPositionAngle().toRadians();
-        t = t + getRotationConfig(ctx.getIssPort()).toRadians().getMagnitude();
-        AffineTransform rotWithPosAngle = AffineTransform.getRotateInstance(-t);
-        AffineTransform rotAgainstPosAngle = AffineTransform.getRotateInstance(t);
+        final double t = ctx.getPositionAngle().toRadians()
+                + getRotationConfig(ctx.getIssPort()).toRadians().getMagnitude();
+        final AffineTransform rotWithPosAngle = AffineTransform.getRotateInstance(-t);
+        final AffineTransform rotAgainstPosAngle = AffineTransform.getRotateInstance(t);
 
         // Get the valid range of the CWFS 3 probe at all offset positions and
         // compute the center of that range.
-        Area p3range = new Area(probeRange3(ctx));
-        Rectangle2D b = p3range.getBounds2D();
-        Point2D center = new Point2D.Double(b.getCenterX(), b.getCenterY());
+        final Area p3range = new Area(probeRange3(ctx));
+        final Rectangle2D b = p3range.getBounds2D();
+        final Point2D center = new Point2D.Double(b.getCenterX(), b.getCenterY());
 
         // We have to translate the rectangle.  If there is a primary CWFS3
         // star, then we translate the rectangle to there.  Otherwise, we assume
         // the center of the probe 3 range computed above.
-        Point2D xlat = new Point2D.Double(center.getX(), center.getY());
-
-        // Find the cwfs3 star, if any.
-        Option<Point2D> cwfs3Opt = getPrimaryCwfs3Offset(ctx);
-        if (!cwfs3Opt.isEmpty()) {
-            Point2D cwfs3 = cwfs3Opt.getValue();
-
+        final Point2D xlat = getPrimaryCwfs3Offset(ctx).map(cwfs3 -> {
```

Comment:
Again, this is just a cleaner way of handling the calculations. There was no reason to create `xlat` twice when only one of the two is used.